pkg/core/blockchain.go

```diff
@@ -731,17 +733,35 @@ func (bc *Blockchain) jumpToStateInternal(p uint32, stage stateChangeStage) erro
 	default:
 		return fmt.Errorf("unknown state jump stage: %d", stage)
 	}
-	block, err := bc.dao.GetBlock(bc.GetHeaderHash(p + 1))
-	if err != nil {
-		return fmt.Errorf("failed to get block to init MPT: %w", err)
+	var root util.Uint256
+	if bc.config.NeoFSStateSyncExtensions {
+		var chkp statesync.Checkpoint
+		data, err := bc.dao.Store.Get([]byte{byte(storage.SYSStateSyncCheckpoint)})
+		if err != nil {
+			return fmt.Errorf("failed to get checkpoint metadata: %w", err)
+		} else {
+			br := io.NewBinReaderFromBuf(data)
+			chkp.DecodeBinary(br)
+			if br.Err != nil {
+				return fmt.Errorf("failed to decode checkpoint metadata: %w", br.Err)
+			}
```

Comment:
Oh-oh, this code is repeated at least two times. We need to move this code to a helper and reuse it everywhere (along with `PutStateSyncCheckpoint`). Given the fact that Checkpoint is already leaking from statesync module and we already have Stateroot field exported and used from Blockchain, let's place this helper near `(dao *Simple) GetStateSyncPoint()`, rename the structure to StateSyncCheckpoint and export all its fields.